### PR TITLE
support link text with boolean config

### PR DESCRIPTION
**Pull request by xrutayisire**
**Resolves**:  [DT-2261](https://linear.app/prismic/issue/DT-2261/aaprismicdev-prismic-mock-supports-text-in-link-fields)

### Description

- Update to support allowText instead of text config with Key Text.

### Checklist

- [x] If my changes require tests, I added them.
- [x] If my changes affect backward compatibility, it has been discussed.
- [ ] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

File: src/lib/buildContentRelationshipField.ts [hunk 1]
```diff
@@ -4,7 +4,6 @@ type BuildContentRelationshipFieldConfig<
 	Document extends prismic.PrismicDocument,
 > = {
 	document: Document;
-	text?: string;
```
Comment (angeloashmore) [inline src/lib/buildContentRelationshipField.ts:7]: Related question from `@prismicio/client`: https://github.com/prismicio/prismic-client/pull/352#discussion_r1758100722

**Review by angeloashmore — COMMENTED** [review]
Approval is pending a question from `@prismicio/client`: https://github.com/prismicio/prismic-client/pull/352#discussion_r1758100722

File: src/lib/buildContentRelationshipField.ts (hunk 1 above)
Comment (dani-mp) [reply src/lib/buildContentRelationshipField.ts:7]: @angeloashmore can we merge this one into the main one?

Comment (angeloashmore) [reply src/lib/buildContentRelationshipField.ts:7]: Yes, I'll merge and give another review with everything together.

**Pull request merged**